src/main/java/uk/gov/mca/beacons/service/registrations/GetAllBeaconsService.java

```diff
@@ -43,6 +44,25 @@ public GetAllBeaconsService(
     final List<Beacon> allBeacons = beaconRepository.findAll();
     if (allBeacons.size() == 0) return emptyList();
 
+    return getMappedBeacons(allBeacons);
+  }
+
+  public Beacon find(UUID id) {
+    final Optional<Beacon> beacon = beaconRepository.findById(id);
+
+    return beacon
+      .map(
+        foundBeacon -> {
+          List<Beacon> beaconInList = new ArrayList<Beacon>();
+          beaconInList.add(foundBeacon);
+
+          return getMappedBeacons(beaconInList).get(0);
+        }
+      )
+      .orElse(new Beacon());
```

Comment:
could this return null rather than new Beacon and the endpoint 404 when no beacon with UUID is found?

Reply:
Yes, realised this as I was falling asleep 😂 But I think I've sorted it out with the latest commit!